plugins/resources/src/unixfilesystem.cpp

```diff
@@ -83,6 +87,198 @@
 const std::string DEFAULT_VAULT_DIR_MODE( "default_vault_directory_mode_kw" );
 const std::string HIGH_WATER_MARK( "high_water_mark" ); // no longer used
 const std::string REQUIRED_FREE_INODES_FOR_CREATE("required_free_inodes_for_create"); // no longer used
+const std::string HOST_MODE("HOST_MODE");
+const std::string HOST_LIST("HOST_LIST");
+
+namespace {
+    bool is_operating_in_detached_mode(irods::plugin_property_map& prop_map)
+    {
+
+        std::string host_mode_str;
+
+        irods::error ret = prop_map.get< std::string >(HOST_MODE, host_mode_str);
+        if (ret.ok()) {
+            if (boost::iequals(host_mode_str.c_str(), "detached"))  {
+                return true;
+            }
+        }
+
+        // default is attached mode
+        return false;
+    }
+}
+
+static std::vector<std::string> split(const std::string str, char delim)
+{
+    std::vector<std::string> result;
+    std::istringstream ss{str};
+    std::string token;
+    while (std::getline(ss, token, delim)) {
+        if (!token.empty()) {
+            result.push_back(token);
+        }
+    }
+    return result;
+}
+
+// returns a tuple with the following:
+//   bool - true iff the resource_hostname is in HOST_LIST string
+//   std::optional<std::string> - the new vault path if it is defined
+std::tuple<bool, std::optional<std::string>> get_detached_mode_vault_path(
+        irods::plugin_property_map& prop_map,
+        const std::string& resource_hostname,
+        const std::string& resource_name)
+{
+    using logger = irods::experimental::log;
+
+    bool in_host_list = false;
+    std::string new_vault_path;
+
+    std::string host_list_str;
+
+    irods::error ret = prop_map.get< std::string >(HOST_LIST, host_list_str);
+    if (!ret.ok()) {
+
+        // no HOST_LIST parameter, all hosts assumed to be able to handle request
+        // and original vault path used for all hosts
+        in_host_list = true;
+        return std::make_tuple(in_host_list, std::nullopt);
+    }
+
+    // have a HOST_LIST parameter
+
+    // split parameter by delimiter (,)
+    char delimiter = ',';
+    std::string token;
+
+    std::string resource_hostname_with_colon = resource_hostname + ":";
+
+    std::vector<std::string> tokens = split(host_list_str, delimiter);
+
+    for (std::string token : tokens) {
```

Comment:
`clazy` suggests a change here:

> Missing reference in range-for with non trivial type (`std::string`) [`clazy-range-for-loop-reference`]

```suggestion
    for (std::string& token : tokens) {
```

Reply:
Done.  Also added a const qualifier.